storage/range.go

```diff
@@ -774,6 +783,19 @@ func (r *Range) executeCmd(method string, args proto.Request, reply proto.Respon
 		return util.Errorf("unrecognized command %q", method)
 	}
 
+	// If we are applying a raft command, update the applied index.
```

Comment:
That's what I originally did, but I think it's better to keep it separate. If there's an error and we don't commit, so be it, but if anything gets committed we need to record this index. I think we need to rework error handling here to distinguish transient/node-local errors (like a disk filling up, which needs to mark the node as dead to prevent the state machines from diverging) from permanent/deterministic errors (where every node will produce the same error so the entry can still be considered committed/applied).